src/PrMerger.js

```diff
@@ -6,14 +6,44 @@ const GH = require('./GitHubUtil.js');
 const Util = require('./Util.js');
 const MergeContext = require('./MergeContext.js');
 
+class PrScanResult {
+    constructor(prs) {
+        this.scanDate = new Date(); // the scan starting time
+        this.awakePrs = [...prs]; // PRs that were not delayed during the scan
+    }
+
+    isStillUnchanged(freshRawPr, freshScanDate) {
+        const savedRawPr = this.awakePrs.find(el => el.number === freshRawPr.number);
+        if (!savedRawPr)
+            return false; // this scan has not seen freshRawPR
+        if (savedRawPr.updated_at !== freshRawPr.updated_at)
+            return false; // PR has changed since this scan
+        // treat recently updated PRs as changed PRs thus factoring in a
+        // (slight) possibility of same-timestamp changes
+        const unmodifiedDurationMs = freshScanDate - new Date(savedRawPr.updated_at);
+        return unmodifiedDurationMs > 1000*3600;
+    }
+
+    forgetPr(rawPr) {
+        assert(this.awakePrs.find(el => el.number === rawPr.number));
+        this.awakePrs = this.awakePrs.filter(el => el.number !== rawPr.number);
```

Comment:
Just to speed it up a little:

```suggestion
        const oldPrs = this.awakePrs;
        this.awakePrs = this.awakePrs.filter(el => el.number !== rawPr.number);
        assert(oldPrs.length === this.awakePrs.length + 1); // one PR was removed
```